test/node/connection-handshake.js

```diff
@@ -0,0 +1,96 @@
+'use strict';
+
+const test = require('tap');
+
+const jstp = require('../..');
+
+const app = require('../fixtures/application');
+const application = new jstp.Application(app.name, app.interfaces);
+
+const Transport = require('../unit/mock/transport');
+
+let server;
+let client;
+
+test.beforeEach((done) => {
+  const port = 3333;
```

Comment:
This is prone to a race condition. The server isn't guaranteed to be shut down before next test starts. However, it is unlikely to happen because of small number of connections, so let's fix it after https://github.com/metarhia/jstp/pull/190 lands.